src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+### Index Access
+
+To access an element in an array by its index, use square brackets `[]` with the index inside. For example let's say we have a `school` record with some student results. 
+
+```surql title="Query"
+CREATE student SET results = [
+	{ score: 76, date: "2017-06-18T08:00:00Z", name: "Algorithmics" },
+	{ score: 83, date: "2018-03-21T08:00:00Z", name: "Concurrent Programming" },
+	{ score: 69, date: "2018-09-17T08:00:00Z", name: "Advanced Computer Science 101" },
+	{ score: 73, date: "2019-04-20T08:00:00Z", name: "Distributed Databases" },
+];
+```
+
+```json title="Response"
+[
+	{
+		id: student:urxaykt4qkbr8rs2o68j,
+		results: [
+			{
+				date: '2017-06-18T08:00:00Z',
+				name: 'Algorithmics',
+				score: 76
+			},
+			{
+				date: '2018-03-21T08:00:00Z',
+				name: 'Concurrent Programming',
+				score: 83
+			},
+			{
+				date: '2018-09-17T08:00:00Z',
+				name: 'Advanced Computer Science 101',
+				score: 69
+			},
+			{
+				date: '2019-04-20T08:00:00Z',
+				name: 'Distributed Databases',
+				score: 73
+			}
+		]
+	}
+]
+```
+
+To access the first student in the `results` array, you can use the following idiom:
+
+
+```surql
+SELECT results[0].score FROM student;
+```
+
+```json title="Response"
+[
+  {
+    "results": [
+      { "score": 76 }
+    ]
+  }
+]
+```
+
+Here, `results[0].score` accesses the score of the first student in the `results` array. 
+
+### All Elements
+
+To access all elements in an array or all fields in an object, use `[*]` or simply `*`. This is useful when you want to access all the elements in an array or all the fields in an object. 
+
+```surql
+SELECT results[*].score FROM student;
+```
+
+```json title="Response"
+[
+	{
+		results: {
+			score: [
+				76,
+				83,
+				69,
+				73
+			]
+		}
+	}
+]
+```
+
+This idiom selects all elements in the `score` array.
+
+### Last Element
+
+Addionally to access the last element of an array, use `[$]`. Refereing to the `student` record above, we can access the last element of the `results` array using the following idiom:
+
+```surql
+SELECT results[$].score FROM student;
+```
+```json title="Response"
+[
+	{
+		results: {
+			score: 73
+		}
+	}
+]
+```
+
+This idiom accesses the last element of the `score` array.
+
+### First Element
+
+As we saw initially to access the first element of an array, use `[0]`.
+
+```surql
+SELECT results[0].score FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "results": [
+      { "score": 76 }
+    ]
+  }
+]
+```
+
+### Flatten
+
+To flatten nested arrays, use `...` (ellipsis). This is useful when you want to access all the elements in a nested array. 
+
+```surql
+SELECT results.score... FROM student;
+```
+
+This idiom flattens any nested arrays within `results.score`, returning a single-level array.
+
+### Where Filter
+
+You can also filter elements based on a condition, use `[WHERE ... ]`. This will only return the elements that match the condition.
```

Comment:
Recommend mentioning `$this` here which is needed if there isn't a field name to work on.

[1,2,3][WHERE > 3]; -- doesn't work
[1,2,3][WHERE $this > 3]; -- works

[{data: 1}, {data: 2}][WHERE data > 0];
[{data: 1}, {data: 2}][WHERE $this.data > 0];

Probably also a note about how `?` is also an alias for `WHERE` because a lot of people seem to enjoy using that short syntax.